bundles/org.openhab.binding.tellstick/src/main/java/org/openhab/binding/tellstick/internal/local/TelldusLocalBridgeHandler.java

```diff
@@ -0,0 +1,313 @@
+import org.openhab.core.types.RefreshType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.tellstick.device.TellstickDeviceEvent;
+import org.tellstick.device.TellstickException;
+import org.tellstick.device.iface.Device;
+
+/**
+ * {@link TelldusLocalBridgeHandler} is the handler for Telldus Local API (Tellstick ZNET v1/v2) and connects it
+ * to the framework. All {@link TelldusDevicesHandler}s use the
+ * {@link TelldusLocalDeviceController} to execute the actual commands.
+ *
+ * @author Jan Gustafsson- Initial contribution
+ */
+public class TelldusLocalBridgeHandler extends BaseBridgeHandler implements TelldusBridgeHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(TelldusLocalBridgeHandler.class);
+
+    private TellstickLocalDevices deviceList = null;
+    private TellstickLocalSensors sensorList = null;
+    private TelldusLocalDeviceController controller = null;
+    private List<DeviceStatusListener> deviceStatusListeners = new Vector<>();
+    private final HttpClient httpClient;
+
+    private static final int REFRESH_DELAY = 10;
+
+    public TelldusLocalBridgeHandler(Bridge bridge, HttpClient httpClient) {
+        super(bridge);
+        this.httpClient = httpClient;
+    }
+
+    private ScheduledFuture<?> pollingJob;
+    private ScheduledFuture<?> immediateRefreshJob;
+
+    @Override
+    public void dispose() {
+        logger.debug("Local Handler disposed.");
+        if (pollingJob != null) {
+            pollingJob.cancel(true);
+        }
+        if (this.controller != null) {
+            this.controller.dispose();
+        }
+        deviceList = null;
+        sensorList = null;
+        super.dispose();
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("Initializing Telldus Local bridge handler.");
+        TelldusLocalConfiguration configuration = getConfigAs(TelldusLocalConfiguration.class);
+        this.controller = new TelldusLocalDeviceController(configuration, httpClient);
+        startAutomaticRefresh(configuration.refreshInterval);
+        refreshDeviceList();
+        updateStatus(ThingStatus.ONLINE);
+    }
+
+    private synchronized void startAutomaticRefresh(long refreshInterval) {
+        if (pollingJob != null && !pollingJob.isCancelled()) {
+            pollingJob.cancel(true);
+        }
+        pollingJob = scheduler.scheduleWithFixedDelay(this::refreshDeviceList, 0, refreshInterval,
+                TimeUnit.MILLISECONDS);
+    }
+
+    private void scheduleImmediateRefresh() {
+        // We schedule in 10 sec, to avoid multiple updates
+        logger.debug("Current remaining delay {}", pollingJob.getDelay(TimeUnit.SECONDS));
+        if (pollingJob.getDelay(TimeUnit.SECONDS) > REFRESH_DELAY) {
+            if (immediateRefreshJob == null || immediateRefreshJob.isDone()) {
+                immediateRefreshJob = scheduler.schedule(this::refreshDeviceList, REFRESH_DELAY, TimeUnit.SECONDS);
```

Comment:
You could use the built-in feature by using `ExpiringCache`. See https://openhab.org/javadoc/latest/org/openhab/core/cache/expiringcache

Reply:
I don't understand how this could be done, can you provide an example?

Reply:
You could take a look at some other bindings, using `ExpiringCache`: https://github.com/openhab/openhab-addons/search?q=expiringcache It's not really hard to use it.

Reply:
OK; I made a switch to using ExpiringCache.